packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -220,10 +220,8 @@ export async function enhanceConfigsForDestination(
         if (destination) {
             // in case of a full url destination remove the path defined in the destination from the forwarded call
             if (isFullUrlDestination(destination)) {
-                const destPath = new URL(destination.Host).pathname.replace(/\/$/, '');
-                if (backend.path.startsWith(destPath) && !backend.pathReplace) {
-                    backend.pathReplace = backend.path.replace(destPath, '');
-                }
+                backend.path = new URL(destination.Host).pathname.replace(/\/$/, '');
+                backend.pathReplace = backend.pathReplace ?? '/';
```

Comment:
Could you explain the change? If I look at the two implementations the behavior changes for two cases
1. if `pathReplace` is undefined/null you now set it to `/`.
2. if the `pathReplace` is undefined and the `path` contains the path of of the full url destinations, then `pathReplace` is used to make it work

(1) surprises me because, I'd expect that we don't do anything if `pathReplace` is undefined and (2) seemed logical to me, and I am confused that it has been removed.

Reply:
Sure, I will try to explain the best I could. Let's say that we have a full url destination with the following URL `https://services.odata.org/V4/Northwind/Northwind.svc/`. When an application is generated then in the `manifest.json` we will have as OData URI `/V4/Northwind/Northwind.svc/` and in the `ui5.yaml` we will have as `/V4` as `path` for the `backend-proxy-middleware`.
Now when the application is started all backend request are starting with `/V4/Northwind/Northwind.svc/`, but since we have a full url destination this part of the request needs to be deleted. So I am setting the `backend.path = /V4/Northwind/Northwind.svc/` and `backend.pathReplace = '/'` and now e.g. when a request like this comes `/V4/Northwind/Northwind.svc/$metadata` it will be rewritten just to `/$metadata`.


> (2) seemed logical to me, and I am confused that it has been removed.
I removed it because the check `backend.path.startsWith(destPath)` always returns false, because `backend.path=/V4` and `destPath=/V4/Northwind/Northwind.svc/`, so `pathReplace` is not set the and the `pathReplace` logic is not triggering at all.

Reply:
Thanks a lot!